test/Microsoft.VisualStudio.Threading.Tests/JoinableTaskContextTests.cs

```diff
@@ -45,6 +45,55 @@ public void IsWithinJoinableTask()
         });
     }
 
+    [Fact]
+    public void IsMainThreadBlockedByAnyone()
+    {
+        Assert.False(this.Context.IsMainThreadBlockedByAnyone);
+        AsyncManualResetEvent mainThreadBlockerEvent = new AsyncManualResetEvent(false);
+        AsyncManualResetEvent backgroundThreadMonitorEvent = new AsyncManualResetEvent(false);
+
+        // Start task to monitor IsMainThreadBlockedByAnyone
+        Task.Run(async () =>
+        {
+            await TaskScheduler.Default.SwitchTo(alwaysYield: true);
+            await mainThreadBlockerEvent.WaitAsync();
```

Comment:
This may deadlock the test. Add a cancellation token.

```suggestion
            await mainThreadBlockerEvent.WaitAsync(this.TimeoutToken);
```